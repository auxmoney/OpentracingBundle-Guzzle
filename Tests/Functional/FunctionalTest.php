<?php

declare(strict_types=1);

namespace Auxmoney\OpentracingGuzzleBundle\Tests\Functional;

use Auxmoney\OpentracingBundle\Tests\Functional\JaegerWebFunctionalTest;
use Symfony\Component\Process\Process;

class FunctionalTest extends JaegerWebFunctionalTest
{
    /**
     * @dataProvider provideProjectSetups
     */
    public function testNestedSpansAndHeaderPropagation(string $projectSetup): void
    {
        $this->setUpTestProject($projectSetup);

        $p = new Process(['symfony', 'console', 'test:guzzle'], 'build/testproject');
        $p->mustRun();
        $output = $p->getOutput();
        $traceId = substr($output, 0, strpos($output, ':'));
        self::assertNotEmpty($traceId);

        $spans = $this->getSpansFromTrace($this->getTraceFromJaegerAPI($traceId));
        self::assertCount(5, $spans);

        $traceAsYAML = $this->getSpansAsYAML($spans, '[].{operationName: operationName, startTime: startTime, spanID: spanID, references: references, tags: tags[?key==\'http.status_code\' || key==\'command.exit-code\' || key==\'http.url\' || key==\'http.method\' || key==\'auxmoney-opentracing-bundle.span-origin\'].{key: key, value: value}}');
        self::assertStringEqualsFile(__DIR__ . '/FunctionalTest.expected.yaml', $traceAsYAML);
    }

    public function provideProjectSetups(): array
    {
        return [
            'no handler' => ['noHandler'],
            'existing handler stack' => ['existingHandlerStack'],
            'multiple existing handler stacks' => ['existingHandlerStacks'],
        ];
    }
}
