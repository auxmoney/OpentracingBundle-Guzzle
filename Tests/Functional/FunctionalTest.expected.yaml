operationName: 'test:guzzle'
tags:
  -
    key: auxmoney-opentracing-bundle.span-origin
    value: 'core:command'
  -
    key: command.exit-code
    value: 0
children:
  -
    operationName: 'sending HTTP request'
    tags:
      -
        key: auxmoney-opentracing-bundle.span-origin
        value: 'guzzle:request'
      -
        key: http.method
        value: GET
      -
        key: http.status_code
        value: 200
      -
        key: http.url
        value: 'http://localhost:8000/'
    children:
      -
        operationName: 'http://localhost:8000/'
        tags:
          -
            key: auxmoney-opentracing-bundle.span-origin
            value: 'core:request'
          -
            key: http.method
            value: GET
          -
            key: http.url
            value: 'http://localhost:8000/'
        children:
          -
            operationName: 'App\Controller\TestController::index'
            tags:
              -
                key: auxmoney-opentracing-bundle.span-origin
                value: 'core:controller'
              -
                key: http.status_code
                value: 200
            children:
              -
                operationName: 'sending HTTP request'
                tags:
                  -
                    key: auxmoney-opentracing-bundle.span-origin
                    value: 'guzzle:request'
                  -
                    key: http.method
                    value: GET
                  -
                    key: http.status_code
                    value: 200
                  -
                    key: http.url
                    value: 'https://github.com/auxmoney/OpentracingBundle-Guzzle'
